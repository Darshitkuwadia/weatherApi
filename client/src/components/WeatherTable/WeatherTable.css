.weather-table {
  margin-top: 50px;
  display: block;
  font-family: sans-serif;
  -webkit-font-smoothing: antialiased;
  font-size: 115%;
  overflow: auto;
  width: auto;
}
.weather-table  th {
  background-color: var(--color-brand);
  color: var(--color-white);
  font-weight: normal;
  padding: 10px 20px;
  text-align: center;
}
.weather-table  td {
  background-color: var(--color-grey1);
  color: var(--color-grey3);
  padding: 15px 25px;
}

.weather-icon {
  height: 50px;
  width: 100px;
}
