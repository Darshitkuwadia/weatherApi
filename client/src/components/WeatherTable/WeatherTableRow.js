import React from 'react';
import Skycons from 'react-skycons'

function WeatherTableRow(props) {
  const {
    city,
    weatherIcon,
    time,
    summary,
    temperature
  } = props
  
  let currentTime = new Date(time* 1000)
  
  return (
    <tr>
      <td>{city.charAt(0).toUpperCase() + city.slice(1)}</td>
      <td>{temperature}&#8457;</td>
      <td>{summary}</td>
      <td>{currentTime.toDateString()}</td>
      <td className="weather-icon">
        <Skycons color='black' icon={weatherIcon.toUpperCase().replace(/-/g, '_')}/>
      </td>
    </tr>
  )
}

export default WeatherTableRow