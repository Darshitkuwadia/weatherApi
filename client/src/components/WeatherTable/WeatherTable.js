import React from 'react';

import './WeatherTable.css';
import WeatherTableRow from './WeatherTableRow'

function WeatherTable(props) {
  const {
    children
  } = props
  
  return (
    <table className="weather-table">
      <tbody>
        <tr>
          <th>City</th>
          <th>Temperature</th>
          <th>Summary</th>
          <th>Date</th>
          <th>Weather Icon</th>
        </tr>
        {children}
      </tbody>
    </table>
  )
}

WeatherTableRow.displayName = 'WeatherTable.Row'

WeatherTable.Row = WeatherTableRow

export default WeatherTable