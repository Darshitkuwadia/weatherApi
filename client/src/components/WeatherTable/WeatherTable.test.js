import React from 'react';
import { shallow } from 'enzyme';
import WeatherTable from './WeatherTable';

const city = "WeatherTableRow city"
const weatherIcon = "clear-night"
const time = "WeatherTableRow time"
const summary = "WeatherTableRow summary"
const temperature = "WeatherTableRow temperature"

const weatherTableRow = (
  <WeatherTable.Row
    city={city}
    weatherIcon={weatherIcon}
    time={time}
    summary={summary}
    temperature={temperature}
  />
)

describe('WeatherTable', () => {
  it('should render the WeatherTable', () => { 
    let weatherTable = shallow(
      <WeatherTable/>
    )  
    expect(weatherTable).toMatchSnapshot();
  });
  it('renders with WeatherTable.Row children', () => {
    let weatherTable = shallow(
      <WeatherTable>
        {weatherTableRow}
      </WeatherTable>
    ); 
    expect(weatherTable).toMatchSnapshot();
  });
  describe('<WeatherTable.Row />', () => {
    it('renders', () => {
      const renderedWeatherTableRow = shallow(weatherTableRow)
      expect(renderedWeatherTableRow).toMatchSnapshot()
    })
  })
}); 