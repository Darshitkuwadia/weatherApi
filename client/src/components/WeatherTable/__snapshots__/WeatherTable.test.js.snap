// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherTable <WeatherTable.Row /> renders 1`] = `
<tr>
  <td>
    WeatherTableRow city
  </td>
  <td>
    WeatherTableRow temperature
    â„‰
  </td>
  <td>
    WeatherTableRow summary
  </td>
  <td>
    Invalid Date
  </td>
  <td
    className="weather-icon"
  >
    <ReactSkycons
      autoplay={true}
      color="black"
      icon="CLEAR_NIGHT"
    />
  </td>
</tr>
`;

exports[`WeatherTable renders with WeatherTable.Row children 1`] = `
<table
  className="weather-table"
>
  <tbody>
    <tr>
      <th>
        City
      </th>
      <th>
        Temperature
      </th>
      <th>
        Summary
      </th>
      <th>
        Date
      </th>
      <th>
        Weather Icon
      </th>
    </tr>
    <WeatherTable.Row
      city="WeatherTableRow city"
      summary="WeatherTableRow summary"
      temperature="WeatherTableRow temperature"
      time="WeatherTableRow time"
      weatherIcon="clear-night"
    />
  </tbody>
</table>
`;

exports[`WeatherTable should render the WeatherTable 1`] = `
<table
  className="weather-table"
>
  <tbody>
    <tr>
      <th>
        City
      </th>
      <th>
        Temperature
      </th>
      <th>
        Summary
      </th>
      <th>
        Date
      </th>
      <th>
        Weather Icon
      </th>
    </tr>
  </tbody>
</table>
`;
